AutomaticQC/SpikeClassifiers/imosSpikeClassifierBurstRunningStats.m

```diff
@@ -0,0 +1,74 @@
+function [spikes] = imosSpikeClassifierBurstRunningStats(bindexes, signal, scalefun, dispersionfun, dispersion_factor)
+% function [spikes] = imosSpikeClassifierBurstRunningStats(bindexes, signal, scalefun, dispersionfun, dispersion_factor)
+%
+% A simple thresholding method based on user defined deviations around a central scale. Values above
+% the dispersion will be tagged.
+%
+% Inputs:
+%
+% bindexes - a cell with the invidivudal burst range indexes.
+% signal - A Burst 1-d signal.
+% scalefun - the function to estimate the centre of the distribution. (e.g. mean or median)
+%          Default: @mean
+% dispersionfun - the function to estimate the dispersion (e.g. std, mean_absolute_deviation, median_absolute_deviation)
+%               Default: @std
+% dispersion_factor - the multiplying factor for the dispersion.
+%                   Default: 2
+%
+% Outputs:
+%
+% spikes - An array with spikes indexes.
```

Comment:
aren't then booleans in that case

Reply:
no. always indexes.

Reply:
Ow, I just realize there is a caveat here: I initialize spikes as a NaN double, but this is just for allocation and speed.
This array is filled with the  index values where spikes are found in the series, and them I just reduce the results to non-nans and unique values.

Reply:
it's a trick for speed (avoid growing the array) and lowering the memory requirements - in matlab every cast to a variable is a copy, unless some very strict circustances are followed